subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -235,18 +239,52 @@ public void setForkOptions(ForkOptions forkOptions) {
 
     /**
      * Returns the bootstrap classpath to be used for the compiler process. Defaults to {@code null}.
+     *
+     * @deprecated Use {@link #getBootstrapClasspath()} instead.
      */
-    @Input
-    @Optional
+    @Deprecated
+    @Internal
     public String getBootClasspath() {
+        if (bootClasspath == null && bootstrapClasspath != null) {
+            return bootstrapClasspath.getAsPath();
+        }
         return bootClasspath;
```

Comment:
Did you want to use `DeprecationLogger` here?

Reply:
Added